marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -82,49 +90,229 @@ renderCurrentState walletDetails state =
                       [ text "1hr 2mins left" ]
                   ]
               ]
-          , div [ classNames [ "py-2" ] ]
-              [ renderTasks walletDetails state
-              ]
+          , if currentTab == Tasks then
+              renderTasks state
+            else
+              renderBalances state
           ]
       ]
 
-renderTasks :: forall p. WalletDetails -> State -> HTML p Action
-renderTasks walletDetails state =
+-- This helper function expands actions that can be taken by anybody,
+-- then groups by participant and sorts it so that the owner starts first and the rest go
+-- in alphabetical order
+expandAndGroupByRole ::
+  Maybe Party ->
+  Set Party ->
+  Array NamedAction ->
+  Array (Tuple Party (Array NamedAction))
```

Comment:
Nested `/\` notation can be used in types and patterns too. Just out of curiosity, is there a reason you use `Tuple` in some places and `/\` in others?

Reply:
It's a matter of taste I guess, I like to have the type as Tuple and the value as `/\` but I don't feel to strong for it. 

Also, in places like this I think the verbose version is more clear

```purescript
accounts :: Array (Tuple (Tuple Party Token) BigInteger)
accounts = Map.toUnfoldable $ state ^. (_executionState <<< _state <<< _accounts)
```

The problem here is that the nesting happens in the first value of the tuple and `/\` expects the opposite. So actually doing the pattern match looks a little bit ugly :/

```purescript
<#> ( \((party /\ token) /\ amount) ->
```

@merivale what do you think?

Reply:
I'm with Hernan, I like `Tuple` for the types and `/\` for the value. Couldn't tell you why though... 🤷